service/controller/resource/nodepool/create.go

```diff
@@ -62,16 +63,15 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		}
 	}
 
-	err = r.CtrlClient.Status().Update(ctx, &azureMachinePool)
```

Comment:
This was unnecessary. The variable `azureMachinePool` won't contain local changes made in the transition functions anyway.